File: Garcia-Bautista-Jonathan_Ulises/Pipes/C++/pipes.cpp
#include <iostream>
#include <unistd.h>

using namespace std; 

class PipeWriter {
private:
    int fd[2]; 
    const char* buf; 

public:
        PipeWriter(const char* buffer) {
        buf = buffer;
        if (pipe(fd) < 0) {
            cerr << "Error en la creación del pipe" << endl;
            exit(1);
        }
    }
    void writeToPipe() {
        int i = 0;
        while (true) {
            write(fd[1], buf, 1);
            cout << i++ << " ";
            flush(cout);
        }
    }
    ~PipeWriter() {
        close(fd[0]);
        close(fd[1]);
    }
};

int main() {
    const char* buffer = "a";
    PipeWriter writer(buffer);
    writer.writeToPipe();
    return 0;
}

File: Garcia-Bautista-Jonathan_Ulises/Pipes/Python/pipes.py
import os
import sys
import struct

# python ./pipes.py

def main():
    fd = os.pipe()
    i = 0

    try:
        while True:
            buf = struct.pack('i', i)
            os.write(fd[1], buf)
            print(i, end=' ', flush=True)
            i += 1
    except KeyboardInterrupt:
        print("\nPrograma interrumpido")
        sys.exit(0)

if __name__ == "__main__":
    main()
